tests/integration/path_permissions.go

```diff
@@ -38,27 +45,41 @@ func (s *MountPathPermissionsSuite) waitForPods(labelPattern string) {
 		LabelSelector: labelPattern,
 	}
 	require.NoError(
-		kube.WaitForAnyPodExists(ctx, kube.NewTestClient(""), opts),
+		kube.WaitForAnyPodExists(ctx, s.tc, opts),
 		"pod does not exist",
 	)
 	require.NoError(
-		kube.WaitForAnyPodReady(ctx, kube.NewTestClient(""), opts),
+		kube.WaitForAnyPodReady(ctx, s.tc, opts),
 		"pod not ready",
 	)
 }
 
 func (s *MountPathPermissionsSuite) SetupSuite() {
+	s.testID = generateTestID()
+	s.T().Logf("test ID: %s", s.testID)
 	s.tc = kube.NewTestClient("")
 	require := s.Require()
 
 	// Create smbshare with Spec.Storage.PVC.Path specified
-	createFromFiles(s.defaultContext(), require, s.tc, append(s.commonSources, s.smbshareSources...))
-	s.waitForPods(s.serverLabelPattern)
+	ctx := s.defaultContext()
+	createFromFiles(ctx, require, s.tc, s.commonSources)
+	names := createFromFilesWithSuffix(
+		ctx,
+		s.Require(),
+		s.tc,
+		s.smbShareSources,
+		s.testID,
+	)
+	s.Require().Len(names, 1, "expected one smb share resource")
+	s.testShareName = names[0]
+	lbl := fmt.Sprintf(
+		"samba-operator.samba.org/service=%s", s.testShareName.Name)
+	s.waitForPods(lbl)
 }
 
 func (s *MountPathPermissionsSuite) TearDownSuite() {
 	ctx := s.defaultContext()
-	deleteFromFiles(ctx, s.Require(), s.tc, s.smbshareSources)
+	deleteFromFiles(ctx, s.Require(), s.tc, s.smbShareSources)
```

Comment:
```suggestion
	deleteFromFilesWithSuffix(ctx, s.Require(), s.tc, s.smbShareSources, s.testID)
```

Reply:
Gosh, I didn't stop to think my changes were actually the cause. But it makes total sense. Thanks for digging into it and sorry for not noticing that myself!